src/main/java/com/cloudbees/plugins/credentials/cli/ListCredentialsAsXmlCommand.java

```diff
@@ -0,0 +1,62 @@
+package com.cloudbees.plugins.credentials.cli;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.CredentialsStoreAction;
+import com.cloudbees.plugins.credentials.cli.BaseCredentialsCLICommand;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.domains.DomainCredentials;
+import hudson.Extension;
+import hudson.cli.CLICommand;
+import hudson.model.Items;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Extension
+public class ListCredentialsAsXmlCommand extends BaseCredentialsCLICommand {
+
+    @Argument(metaVar = "STORE", usage = "Store ID. For example the Jenkins credentials global store would " +
+            "be identified by \"system::system::jenkins\"",
+            required = true)
+    private CredentialsStore store;
+
+    @Option(name = "--import", usage = "Read credentials as XML from the standard input and create them in the given store." +
+            "Input format is the same as the output generated by this command without enabling this option.")
+    private boolean doImport;
+
+
+    @Override
+    public String getShortDescription() {
+        return "Export and Import credentials as XML";
+    }
+
+    @Override
+    protected int run() throws Exception {
+        if (doImport) {
+            store.checkPermission(CredentialsProvider.UPDATE);
+            List<DomainCredentials> replacement = (List<DomainCredentials>) Items.XSTREAM.unmarshal(safeXmlStreamReader(stdin));
+            for (DomainCredentials domain : replacement) {
+                for (Credentials credentials : domain.getCredentials()) {
+                    store.addDomain(domain.getDomain(), credentials);
+                }
+            }
+            return 0;
+        } else {
+            store.checkPermission(CredentialsProvider.VIEW);
+
+            List<DomainCredentials> existing = new ArrayList<>();
+            List<Domain> domains = store.getDomains();
+            for (Domain domain : domains) {
+                existing.add(new DomainCredentials(domain, store.getCredentials(domain)));
+            }
+            CredentialsStoreAction.SECRETS_REDACTED.toXML(existing, new OutputStreamWriter(stdout, "UTF-8"));
```

Comment:
this should print (output) the number of secrets exported - so you can verify that all are imported correctly later on.

Reply:
Then a wrapper object is needed (downsides of exporting the model 😃). My intention is keep the change as short as possible.

The information is in the XML, I can image a very simple Xpath to count nodes.

Reply:
I forgot there is no possibility to write to a file these days so that stdout can contain messages :-/